jenkins/bokchoy_db_pull_request.py

```diff
@@ -0,0 +1,212 @@
+        )
+    return data
+
+
+def _file_has_changed(repository, local_file, file_path):
+    """
+    Checks to see if a file differs from the current version
+    in a repo.
+    Returns true if the files are different.
+    """
+    try:
+        repo_file_contents = repository.get_file_contents(file_path).decoded_content
+    except:
+        raise StandardError(
+            "Could not get contents of file: {}".format(file_path)
+        )
+    return local_file != repo_file_contents
+
+
+def _create_branch(repository, branch_name, sha):
+    """
+    Create a new branch with the given sha as its head.
+    """
+    try:
+        repository.create_git_ref(branch_name, sha)
+    except:
+        raise StandardError(
+            "Unable to create git branch: {}. "
+            "Check to make sure this branch doesn't already exist.".format(branch_name)
+        )
+
+
+def _update_file(repository, file_path, commit_message, contents, file_sha, branch_name):
+    """
+    Create a commit on a branch that updates the file_path with the string contents.
+    """
+    try:
+        repository.update_file(file_path, commit_message, contents, file_sha, branch_name)
+    except:
+        raise StandardError(
+            "Error updating database file: {}".format(file_path)
+        )
+
+
+def _create_pull_request(repository, title, body, base, head):
+    """
+    Create a new pull request with the changes in head.
+    """
+    try:
+        pull_request = repository.create_pull(
+            title=title,
+            body=body,
+            base=base,
+            head=head
+        )
+    except:
+        raise StandardError(
+            "Could not create pull request"
+        )
+
+
+@click.command()
+@click.option(
+    '--sha',
+    help="Sha of the merge commit to base the new PR off of",
+    required=True,
+)
+@click.option(
+    '--github_token',
+    help="Github token for authentication",
+    required=True,
+)
+@click.option(
+    '--repo_root',
+    help="Path to local edx-platform repository that will "
+         "hold updated database files",
+    required=True,
+)
+def main(sha, github_token, repo_root):
+    # Connect to github
+    logger.info("Authenticating with Github")
+    github_instance = _authenticate_with_github(github_token)
+    repository = _connect_to_repo(github_instance, "edx-platform")
+
+    # Iterate through the db files and update them accordingly
+    branch_created = False
+    for db_file in BOKCHOY_DB_FILES:
+        repo_file_path = os.path.join('/', DB_CACHE_FILEPATH, db_file)
+
+        # Get the sha of the file in the repository
+        file_sha = _get_file_sha(repository, repo_file_path)
+
+        # Read the local db files that were updated by paver
+        local_file_path = os.path.join(repo_root, DB_CACHE_FILEPATH, db_file)
+        local_file_data = _read_file_contents(local_file_path)
+
+        # Check if the local file is different from whats in the repo currently
```

Comment:
I found this part a little confusing. Perhaps there is a simpler way to detect if there is a change locally, without needing to communicate with the repository and read the files. Git already does the work for checking if HEAD is dirty. When you were investigating PyGithub, did you see anyway to access functionality like 'git status' or 'git ls-files'?

Reply:
For things like comparing, it seemed to be dependent on 2 sha's in a repo. I couldn't find much other than that.